### PR TITLE
Xerces 3.2.3 win ec7

### PR DESCRIPTION
**Pull request by MilesZhang-Jie**

### CODE REVIEW

File: src/stricmp.c [hunk 1]
```diff
@@ -34,7 +34,8 @@ int stricmp(const char* str1, const char* str2)
 #if HAVE_STRCASECMP
 	return strcasecmp(str1, str2);
 #else
-	#error Need implementation of stricmp compatibility function
+	// #error Need implementation of stricmp compatibility function
```
Comment (lucabonifacio) [inline src/stricmp.c]:
```suggestion
```

File: src/strnicmp.c [hunk 2]
```diff
@@ -34,6 +34,7 @@ int strnicmp(const char* str1, const char* str2, size_t count)
 #if HAVE_STRNCASECMP
 	return (count == 0) ? 0 : strncasecmp( str1, str2, count);
 #else
-	#error Need implementation of strnicmp compatibility function
+	// #error Need implementation of strnicmp compatibility function
```
Comment (lucabonifacio) [inline src/strnicmp.c]:
```suggestion
```

File: src/xercesc/internal/XMLScanner.cpp [hunk 3]
```diff
@@ -1273,8 +1273,13 @@ void XMLScanner::scanProlog()
                     if (sawDocTypeDecl) {
                         emitError(XMLErrs::DuplicateDocTypeDecl);
                     }
-
+#ifdef WINCE
+                    // WinCE does NOT support env variables
+                    // Actually XERCES_DISABLE_DTD is not used at all
+                    const char* envvar = NULL; //getenv("XERCES_DISABLE_DTD");
```
Comment (lucabonifacio) [inline src/xercesc/internal/XMLScanner.cpp]:
```suggestion
                    const char* envvar = NULL;
```

File: src/xercesc/util/NetAccessors/WinSock/BinHTTPURLInputStream.cpp [hunk 4]
```diff
@@ -196,32 +196,53 @@ void BinHTTPURLInputStream::Initialize(MemoryManager* const manager)
 #ifdef WITH_IPV6
       gWinsockLib = LoadLibraryA("WS2_32");
 #else
-      gWinsockLib = LoadLibraryA("WSOCK32");
+      gWinsockLib = LoadLibraryW(L"WSOCK32");
```
Comment (lucabonifacio) [inline src/xercesc/util/NetAccessors/WinSock/BinHTTPURLInputStream.cpp]: Isn't this change also affecting other platforms?

File: src/xercesc/util/PlatformUtils.cpp [hunk 5]
```diff
@@ -708,7 +713,10 @@ unsigned long XMLPlatformUtils::getCurrentMillis()
 		ms = (unsigned long)(aTime.time*1000 + aTime.millitm);
 	#else
 		// Make this a warning instead?
-		#error No timing support is configured for this platform. You must configure it.
+		// #error No timing support is configured for this platform. You must configure it.
```
Comment (lucabonifacio) [inline src/xercesc/util/PlatformUtils.cpp]:
```suggestion
```

File: src/xercesc/util/PlatformUtils.cpp [hunk 6]
```diff
@@ -708,7 +713,10 @@ unsigned long XMLPlatformUtils::getCurrentMillis()
 		ms = (unsigned long)(aTime.time*1000 + aTime.millitm);
 	#else
 		// Make this a warning instead?
```
Comment (lucabonifacio) [inline src/xercesc/util/PlatformUtils.cpp]:
```suggestion
```

File: src/xercesc/util/NetAccessors/WinSock/BinHTTPURLInputStream.cpp (hunk 4 above)
Comment (MilesZhang-Jie) [reply src/xercesc/util/NetAccessors/WinSock/BinHTTPURLInputStream.cpp]: You are correct, Luca. This is an issue, need to fix it by using WINCE macro.
#ifdef WINCE
    #ifdef WITH_IPV6
            gWinsockLib = LoadLibraryW(L"WS2_32");
    #else
            gWinsockLib = LoadLibraryW(L"WSOCK32");
    #endif
#else
    #ifdef WITH_IPV6
          gWinsockLib = LoadLibraryA("WS2_32");
    #else
          gWinsockLib = LoadLibraryW(L"WSOCK32");
    #endif
#endif

**Review by lucabonifacio — APPROVED** [review]

**Pull request merged**